src/main/java/com/powsybl/sld/server/utils/CustomVoltageLevelLayoutFactory.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2025, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.sld.server.utils;
+
+import com.powsybl.iidm.network.Connectable;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.TopologyKind;
+import com.powsybl.iidm.network.VoltageLevel;
+import com.powsybl.iidm.network.extensions.BusbarSectionPosition;
+import com.powsybl.iidm.network.extensions.ConnectablePosition;
+import com.powsybl.sld.layout.*;
+import com.powsybl.sld.layout.position.clustering.PositionByClustering;
+import com.powsybl.sld.model.graphs.VoltageLevelGraph;
+
+import java.util.Objects;
+
+/**
+ * @author Kevin Le Saulnier <kevin.le-saulnier at rte-france.com>
+ */
+//TODO: to remove once SmartVoltageLevelLayoutFactory allow us to pass PositionVoltageLevelLayoutFactoryParameters to VoltageLevelLayoutFactory
+public class CustomVoltageLevelLayoutFactory implements VoltageLevelLayoutFactory {
+    private final Network network;
+
+    public CustomVoltageLevelLayoutFactory(Network network) {
+        this.network = Objects.requireNonNull(network);
+    }
+
+    @Override
+    public Layout create(VoltageLevelGraph graph) {
+        VoltageLevel vl = network.getVoltageLevel(graph.getVoltageLevelInfos().getId());
+        return selectVoltageLevelLayoutFactory(vl).create(graph);
+    }
+
+    private VoltageLevelLayoutFactory selectVoltageLevelLayoutFactory(VoltageLevel vl) {
+        PositionVoltageLevelLayoutFactoryParameters positionVoltageLevelLayoutFactoryParameters = defaultPositionVoltageLevelLayoutFactoryParameters();
+
+        return vl.getTopologyKind() == TopologyKind.BUS_BREAKER || hasAtLeastOneExtension(vl)
```

Comment:
I don't get the bus breaker part, I would do the opposite... But in fact it's the same in powsybl-diagram.